### PR TITLE
Implement cascade delete

### LINKED ISSUE #215: Customer Delete should Cascade Delete
Customer Delete should delete all deliveries associated with it

### PR DESCRIPTION
**Pull request by jianyangg**
Deleting customer will delete deliveries
associated with it as well.
Close #215

### CODE REVIEW

**Review by juliusgambe — COMMENTED** [review]
Does this update the json file automatically?

**Comment by jianyangg** [comment]
> Does this update the json file automatically?

Yeap, json file gets updated once command is ran.

File: src/main/java/seedu/address/model/DeliveryBook.java [hunk 1]
```diff
@@ -86,13 +87,25 @@ public void removeDelivery(Delivery key) {
...
+    /**
+     * Removes all deliveries associated with the given customer
+     *
+     * @param customer The customer whose deliveries are to be removed
+     */
+    public void removeDeliveryByCustomer(Customer customer) {
+        // get id of customer, delete deliveries with matching customer id
+        int key = customer.getCustomerId();
+        // concurrent modification exception
+        deliveries.removeIf(delivery -> delivery.getCustomer().getCustomerId() == key);
```
Comment (zhonghan721) [inline src/main/java/seedu/address/model/DeliveryBook.java]: Would it be possible to get the customer id from delivery directly, so as to abide by LoD

**Review by zhonghan721 — COMMENTED** [review]
Some general feedback, otherwise LGTM

File: src/main/java/seedu/address/model/DeliveryBook.java (hunk 1 above)
Comment (jianyangg) [reply src/main/java/seedu/address/model/DeliveryBook.java]: Good spot, done!

**Review by zhonghan721 — APPROVED** [review]
LGTM

**Pull request merged**